### PR TITLE
Added connected.cfg with weights and test data

### PR DESCRIPTION
**Pull request by ashishkrshrivastava**
Relating PR [#16438](https://github.com/opencv/opencv/pull/16438)

### CODE REVIEW

File: testdata/dnn/darknet/generate_darknet_models.py [hunk 1]
```diff
@@ -29,3 +29,4 @@ def genTestData(name, inpShape, outShape):
 genTestData('shortcut_unequal', [1, 2, 3, 5], [1, 4, 3, 5])
 genTestData('shortcut_unequal_2', [1, 2, 3, 5], [1, 2, 3, 5])
 genTestData('convolutional', [1, 3, 4, 6], [1, 5, 2, 3])
+genTestData('connected', [1, 3, 4, 6], [1, 64])
```
Comment (dkurt) [inline testdata/dnn/darknet/generate_darknet_models.py]: Can you please show Darknet's output logs? Just to check if output has `1x64` shape or `1x64x1x1` or `1x1x1x64`

Comment (ashishkrshrivastava) [reply testdata/dnn/darknet/generate_darknet_models.py]: This is how it is showing

```
learning_rate: Using default '0.001000'
momentum: Using default '0.900000'
decay: Using default '0.000100'
subdivisions: Using default '1'
policy: Using default 'constant'
max_batches: Using default '0'
layer     filters    size              input                output
    0 connected                              72  ->    10
    1 dropout       p = 0.50                 10  ->    10
    2 connected                              10  ->     2
Saving weights to connected.weights

```

Comment (ashishkrshrivastava) [reply testdata/dnn/darknet/generate_darknet_models.py]: I have checked by adding convolutional layer too. 
```
[net]
batch=1
channels=3
height=4
width=6

[convolutional]
size=1
stride=1
filters=3
activation=leaky

[connected]
output=10
activation=leaky

[dropout]
probability=.5

[connected]
output=2
activation=leaky

```
Here is the output log
```
layer     filters    size              input                output
    0 conv      3  1 x 1 / 1     6 x   4 x   3   ->     6 x   4 x   3  0.000 BFLOPs
    1 connected                              72  ->    10
    2 dropout       p = 0.50                 10  ->    10
    3 connected                              10  ->     2
```

File: testdata/dnn/darknet/connected.cfg [hunk 2]
```diff
@@ -0,0 +1,16 @@
+[net]
+batch=1
+channels=3
+height=4
+width=6
```
Comment (dkurt) [inline testdata/dnn/darknet/connected.cfg]: Do you mind to choose smaller input resolution? Like `1x2x3x4`?

Comment (ashishkrshrivastava) [reply testdata/dnn/darknet/connected.cfg]: okay.

Comment (ashishkrshrivastava) [reply testdata/dnn/darknet/connected.cfg]: Do you want me to change input for maxpool too ?

Comment (dkurt) [reply testdata/dnn/darknet/connected.cfg]: No, pooling looks fine for me, thanks!

**Pull request merged**